libraries/config_procs.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: config_procs.rb
+
+module CernerSplunk
+  # Module for dynamically generating configuration values via Procs
+  module ConfigProcs
+    # Methods to generate procs for determining the value to be written to a conf file
+    module Value
+      def self.constant(value:, **_)
+        proc { value }
+      end
+
+      def self.vault(coordinate:, default_coords: nil, pick_context: nil, **_)
+        proc { CernerSplunk::DataBag.load coordinate, type: :vault, default: default_coords, pick_context: pick_context }
+      end
+
+      def self.existing(filename:, **_)
+        proc do |context, config|
```

Comment:
Hm... we seem to have changed the API for this proc, Note to self: review changes to application of procs.